subsys/bluetooth/audio/aics.c

```diff
@@ -65,36 +66,33 @@ static ssize_t read_description(struct bt_conn *conn,
 
 #define BT_AICS_SERVICE_DEFINITION(_aics) {                                    \
 	BT_GATT_SECONDARY_SERVICE(BT_UUID_AICS),                               \
-	BT_GATT_CHARACTERISTIC(BT_UUID_AICS_STATE,                             \
-			       BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,        \
-			       BT_GATT_PERM_READ_LESC,                         \
-			       read_aics_state, NULL, &_aics),                 \
-	BT_GATT_CCC(aics_state_cfg_changed,                                    \
-		    BT_GATT_PERM_READ | BT_GATT_PERM_WRITE_LESC),              \
-	BT_GATT_CHARACTERISTIC(BT_UUID_AICS_GAIN_SETTINGS,                     \
-			       BT_GATT_CHRC_READ,                              \
-			       BT_GATT_PERM_READ_LESC,                         \
-			       read_aics_gain_settings, NULL, &_aics),         \
-	BT_GATT_CHARACTERISTIC(BT_UUID_AICS_INPUT_TYPE,                        \
-			       BT_GATT_CHRC_READ,                              \
-			       BT_GATT_PERM_READ_LESC,                         \
-			       read_type, NULL, &_aics),                       \
-	BT_GATT_CHARACTERISTIC(BT_UUID_AICS_INPUT_STATUS,                      \
-			       BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,        \
-			       BT_GATT_PERM_READ_LESC,                         \
-			       read_input_status, NULL, &_aics),               \
-	BT_GATT_CCC(aics_input_status_cfg_changed,                             \
-		    BT_GATT_PERM_READ | BT_GATT_PERM_WRITE_LESC),              \
-	BT_GATT_CHARACTERISTIC(BT_UUID_AICS_CONTROL,                           \
-			       BT_GATT_CHRC_WRITE,                             \
-			       BT_GATT_PERM_WRITE_LESC,                        \
-			       NULL, write_aics_control, &_aics),              \
-	BT_GATT_CHARACTERISTIC(BT_UUID_AICS_DESCRIPTION,                       \
-			       BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,        \
-			       BT_GATT_PERM_READ_LESC,                         \
-			       read_description, NULL, &_aics),                \
-	BT_GATT_CCC(aics_description_cfg_changed,                              \
-		    BT_GATT_PERM_READ | BT_GATT_PERM_WRITE_LESC)               \
+	BT_AUDIO_CHRC(BT_UUID_AICS_STATE,                                      \
+		      BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,                 \
+		      BT_GATT_PERM_READ_ENCRYPT,                               \
```

Comment:
neither `BT_GATT_PERM_READ_LESC` nor `BT_GATT_PERM_WRITE_LESC` are needed, because the access to the characteristic is validated on service level with this patch.

Reply:
Right, but if we set them "correctly" they could be validated at an earlier time at the GATT level instead. It doesn't really make much sense to postpone the LESC check to later (in audio.c) when it can just as easily be done earlier (in gatt.c), which would also make the characteristic declaration better matter the actual security check.

Reply:
But wouldn't `BT_GATT_PERM_WRITE_LESC` exclude OOB method? If I understand correctly, setting `BT_GATT_PERM_WRITE_LESC` would reject access to attribute if not paired with LE Secure Connections.

Reply:
I'm in agreement with you @MariuszSkamra, the `BT_GATT_PERM_WRITE_LESC` would indeed exclude the OOB method.
Furthermore, the OOB method is a very special case (both key size + OOB), which we as a combination are unlikely to get into `gatt.c`.

I think the current approach is the best

Reply:
Ah, right, I see. Good point. I had no considered that adding `BT_GATT_PERM_WRITE_LESC` would prevent OOB pairing in that regard. 

I am also in agreement of this.